### PR TITLE
Move the telemetry collector implementation in vs common

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/1628

Regression? Last working version: RPS regression, but not a functional one.

## Description

When I made the [change](https://github.com/NuGet/NuGet.Client/commit/75b1772e11db31ffbba5c11669aab5a06e2e6ee8) to enable packages.config support in `IVsNuGetProjectUpdateEvents`, I added the `INuGetTelemetryProvider` as a parameter tot he object to ensure that the initialization of it is complete as MEF & VS Package loading are not coordinated. 

The NuGetTelemetryProvider was part of the `NuGet.VisualStudio.Implementation` assembly and this was an assembly that wasn't loaded in the scenario before. 

Note that due to an RPS hick-up this regression was missed at PR time. 

By moving NuGetTelemetryProvider  to NuGet.VisualStudio.Common we ensure that there's no extra assembly loaded in the scenario.

Note that we now have to remove `nameof` for NuGet.VisualStudio.Contracts types in the ExtensibilityTelemetryCollector because NuGet.VisualStudio,Common does not depend on `NuGet.VisualStudio.Contracts`. 
Adding that dependency would cause another RPS regression in the same scenario, which would be `-NuGet.VisualStudio.Implementation`, `+NuGet.VisualStudio.Contracts` so we land here :) 

Note that I'm testing this fix in `https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/399790`. which is based on my build in https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6168319&view=results.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [x] Test exception - There's an existing RPS test.
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/ExtensibilityTelemetryCollector.cs [hunk 1]
```diff
@@ -246,4 +245,14 @@ protected override void OnEventWritten(EventWrittenEventArgs eventData)
             }
         }
     }
+
+    internal static class DeconstructionExtensions
+    {
+        [DebuggerStepThrough]
+        public static void Deconstruct<TKey, TValue>(this KeyValuePair<TKey, TValue> pair, out TKey key, out TValue value)
```
Comment (zivkan) [inline src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/ExtensibilityTelemetryCollector.cs:252]: Why is this needed? 😕

**Review by zivkan — APPROVED** [review]

File: src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/ExtensibilityTelemetryCollector.cs [hunk 2]
```diff
@@ -8,7 +8,6 @@
 using System.Threading;
 using NuGet.Common;
 using NuGet.SolutionRestoreManager;
-using NuGet.VisualStudio.Contracts;
 using NuGet.VisualStudio.Etw;
 
 namespace NuGet.VisualStudio.Telemetry
```
Comment (zivkan) [inline src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/ExtensibilityTelemetryCollector.cs:13]: Is NuGet.VisualStudio.Common's default namespace `NuGet.VisualStudio` or `NuGet.VisualStudio.Common`?

I'd like the namespace to be updated if it has different defaults.

Comment (nkolev92) [reply src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/ExtensibilityTelemetryCollector.cs:13]: It's windly inconsistent. 

There's a lot of `NuGet.VisualStudio` and a lot of `NuGet.VisualStudio.Common`. 

`NuGet.VisualStudio,Telemetry` is also a namespace used in JoinableTaskExtensions and INuGetTelemetryProvider

File: src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/ExtensibilityTelemetryCollector.cs (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/ExtensibilityTelemetryCollector.cs:252]: There's a few `foreach` loops in this class that deconstructs the KVP into a value tuple.

**Comment by nkolev92** [comment]
https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/399790 looks good and fixes the RPS regression.

**Review by jeffkl — APPROVED** [review]
